sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -24,30 +24,33 @@ ctypedef cnp.float64_t DOUBLE
 
 
 def csr_row_norms(X):
-    """L2 norm of each row in CSR matrix X."""
+    """Squared L2 norm of each row in CSR matrix X."""
     if X.dtype not in [np.float32, np.float64]:
         X = X.astype(np.float64)
-    return _csr_row_norms(X.data, X.shape, X.indices, X.indptr)
+    return np.asarray(_csr_row_norms(X.data, X.indices, X.indptr))
```

Comment:
Memoryviews have a `.base` attribute which allows returning the base object, here the underlying NumPy array. This can be used in place of `np.asarray`.

```suggestion
    return _csr_row_norms(X.data, X.indices, X.indptr).base
```